engine/table/src/main/java/io/deephaven/engine/table/impl/util/ServerStateStreamPublisher.java

```diff
@@ -0,0 +1,94 @@
+package io.deephaven.engine.table.impl.util;
+
+import io.deephaven.chunk.WritableChunk;
+import io.deephaven.chunk.attributes.Values;
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.table.impl.sources.ArrayBackedColumnSource;
+import io.deephaven.qst.type.Type;
+import io.deephaven.stream.StreamConsumer;
+import io.deephaven.stream.StreamPublisher;
+import io.deephaven.stream.StreamToBlinkTableAdapter;
+import io.deephaven.time.DateTimeUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Objects;
+
+class ServerStateStreamPublisher implements StreamPublisher {
+
+    private static final TableDefinition DEFINITION = TableDefinition.of(
+            ColumnDefinition.ofTime("IntervalStartTime"),
+            ColumnDefinition.ofInt("IntervalDurationMicros"),
+            ColumnDefinition.ofInt("TotalMemoryMiB"),
+            ColumnDefinition.ofInt("FreeMemoryMiB"),
+            ColumnDefinition.ofShort("IntervalCollections"),
+            ColumnDefinition.ofInt("IntervalCollectionTimeMicros"),
+            ColumnDefinition.ofShort("IntervalUGPCyclesOnBudget"),
+            ColumnDefinition.of("IntervalUGPCyclesTimeMicros", Type.intType().arrayType()),
+            ColumnDefinition.ofShort("IntervalUGPCyclesSafePoints"),
+            ColumnDefinition.ofInt("IntervalUGPCyclesSafePointTimeMicros"));
+
+    private static final int CHUNK_SIZE = ArrayBackedColumnSource.BLOCK_SIZE;
+
+    public static TableDefinition definition() {
+        return DEFINITION;
+    }
+
+    private WritableChunk<Values>[] chunks;
+    private StreamConsumer consumer;
+
+    ServerStateStreamPublisher() {
+        chunks = StreamToBlinkTableAdapter.makeChunksForDefinition(DEFINITION, CHUNK_SIZE);
+    }
+
+    @Override
+    public void register(@NotNull StreamConsumer consumer) {
+        if (this.consumer != null) {
+            throw new IllegalStateException("Can not register multiple StreamConsumers.");
+        }
+        this.consumer = Objects.requireNonNull(consumer);
+    }
+
+    public synchronized void add(
+            final long intervalStartTimeMillis,
+            final int intervalDurationMicros,
+            final int totalMemoryMiB,
+            final int freeMemoryMiB,
+            final short intervalCollections,
+            final int intervalCollectionTimeMicros,
+            final short intervalUGPCyclesOnBudget,
+            final int[] intervalUGPCyclesTimeMicros,
+            final short intervalUGPCyclesSafePoints,
+            final int intervalUGPCyclesSafePointTimeMicros) {
+        chunks[0].asWritableLongChunk().add(DateTimeUtils.millisToNanos(intervalStartTimeMillis));
```

Comment:
This could be more efficient if we did our casts once per buffer, and only set sizes in flushInternal.